include/onnxruntime/core/optimizer/rewrite_rule.h

```diff
@@ -11,31 +11,31 @@ namespace onnxruntime {
-
-When creating a new rewrite rule, two main function have to be implemented: SatisfyCondition and Apply.
-- SatisfyCondition determines whether the rule will be triggered, and can include multiple condition checks.
-It is advisable to add the more selective checks first, because those will lead to discarding fast rules that 
-cannot be applied on a node.
-- Apply is the actual body of the rule that will be executed if the checks in SatisfyCondition are passed
-successfully. Note that additional, more complex checks can be included in the Apply if putting them in the
-SatisfyCondition would lead to duplicate work (e.g., when we make a check on a Node attribute but we need
-that attribute to execute the rule too).
-
-In general, simple fast checks are a better fit for SatisfyCondition, whereas more complex ones can be 
-added in the Apply.
+The base class for a rewrite rule. A rewrite rule represents a semantics-preserving transformation of a 
+computation graph. It can be used to represent, for example, the elimination of operators that serve as 
+no-ops (e.g., dropout during inference), as well as inlining of "function" definitions or the dual operation 
+of replacing a complex expression by an equivalent function-call). Unlike the more general GraphTransformer, 
+a rewrite rule is a more local transformation that is triggered on a particular node of the graph. 
+
+Each rule has a set of conditions and a body. The conditions have to be satisfied for the body of the rule 
+to be triggered. Therefore, when creating a new rewrite rule, two main functions have to be implemented: 
+- SatisfyCondition defines the condition checks. It is advisable to add the more selective checks first, 
+  because those will lead to discarding fast rules that cannot be applied on a node.
+- Apply is the actual body of the rule that will be executed if SatisfyCondition returns true for a particular
+  node. Note that additional, more complex checks can be included in the Apply if putting them in the
+  SatisfyCondition would lead to duplicate work (e.g., when we make a check on a Node attribute but we need
+  that attribute to execute the rule too).
+In general, simple fast checks are a better fit for SatisfyCondition, whereas more complex ones can be added 
+in the Apply.
+
+In order to avoid evaluating the SatisfyCondition for each rule and each node of the graph, each rewrite rule
+should specify the target op types for which a rule will be evaluated, by overriding the TargetOpTypes() function.
+If the op type of a node is not included in the target op types of a rule, that rule would not be considered at all.
+If the list of op types is left empty, that rule will be triggered for every op type.
 */
 class RewriteRule {
  public:
-  RewriteRule(const std::string& name, const std::string& desc)
-      : name_(name), desc_(desc) {
-  }
+  RewriteRule(const std::string& name, const std::string& desc) : name_(name), desc_(desc) {}
```

Comment:
Can the description be removed here? The extra string costs runtime memory and code to generate.

Reply:
Done with last commit. Similarly, I also removed the description of the graph transformers.